lib/scraped_page.rb

```diff
@@ -1,5 +1,74 @@
+# frozen_string_literal: true
 require 'scraped_page/version'
 
-module ScrapedPage
-  # Your code goes here...
+require 'nokogiri'
+require 'open-uri'
+require 'field_serializer'
+require 'scraped_page_archive'
+
+# Abstract class which scrapers can extend to implement their functionality.
+class ScrapedPage
+  include FieldSerializer
+
+  class Response
+    attr_reader :status, :headers, :body
+
+    def initialize(body:, status: 200, headers: {})
+      @status = status
+      @headers = headers
+      @body = body
+    end
+  end
+
+  module Strategy
+    class LiveRequest
+      def response(url)
+        log "Fetching #{url}"
+        response = open(url)
+        {
+          status:  response.status.first.to_i,
+          headers: response.meta,
+          body:    response.read,
+        }
+      end
+
+      private
+
+      def log(message)
+        warn "[#{self.class}] #{message}"
```

Comment:
Ah yeah, this was handy while I was developing, but yep, no need for it to be on by default. I've added a `VERBOSE` environment variable to enable logging in https://github.com/everypolitician/scraped_page/pull/7/commits/6666c84ad12f863a2c142812499d92c2753b09b8.